authentication/src/test/kotlin/com/uber/sdk2/auth/internal/AuthProviderTest.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2024. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.sdk2.auth.internal
+
+import androidx.appcompat.app.AppCompatActivity
+import com.uber.sdk2.auth.RobolectricTestBase
+import com.uber.sdk2.auth.api.PKCEGenerator
+import com.uber.sdk2.auth.api.exception.AuthException
+import com.uber.sdk2.auth.api.request.AuthContext
+import com.uber.sdk2.auth.api.request.AuthDestination
+import com.uber.sdk2.auth.api.request.AuthType
+import com.uber.sdk2.auth.api.request.CrossApp
+import com.uber.sdk2.auth.api.request.PrefillInfo
+import com.uber.sdk2.auth.api.response.AuthResult
+import com.uber.sdk2.auth.api.response.PARResponse
+import com.uber.sdk2.auth.api.response.UberToken
+import com.uber.sdk2.auth.api.sso.SsoLink
+import com.uber.sdk2.auth.internal.service.AuthService
+import com.uber.sdk2.auth.internal.shadow.ShadowSsoConfigProvider
+import com.uber.sdk2.auth.internal.shadow.ShadowSsoLinkFactory
+import com.uber.sdk2.auth.internal.sso.SsoLinkFactory
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.robolectric.annotation.Config
+import org.robolectric.shadow.api.Shadow
+import retrofit2.Response
+
+@Config(shadows = [ShadowSsoLinkFactory::class, ShadowSsoConfigProvider::class])
+class AuthProviderTest : RobolectricTestBase() {
+  private val activity: AppCompatActivity = mock()
+  private val authService: AuthService = mock()
+  private val codeVerifierGenerator: PKCEGenerator = mock()
+
+  private lateinit var ssoLink: SsoLink
+
+  @Before
+  fun setUp() {
+    ssoLink = Shadow.extract<ShadowSsoLinkFactory>(SsoLinkFactory).ssoLink
+  }
+
+  @Test
+  fun `test authenticate when PKCE flow should return tokens`() = runTest {
```

Comment:
AFAIK Robolectric / Android does not support backtick style of test names

Reply:
I think it works fine in unit tests but if you try to write `androidTests` it breaks. Not because of backticks but due to the space in the function name. 
I read that android also allowed adding space in the function names starting API 30
https://android.googlesource.com/platform/art/+/ab5f4c17a86b8c808dba862db566f0ffa1146367